pkgs/tools/audio/botamusique/default.nix

```diff
@@ -59,17 +53,28 @@ stdenv.mkDerivation rec {
     ./no-runtime-update.patch
   ];
 
+  prePatch = ''
+    postPatchHooks=("''${postPatchHooks[@]/npmConfigHook}")
+  '';
```

Comment:
npmSourceRoot when

cc @winterqt @lilyinstarlight

Reply:
Yeah, that's ugly, yikes.

Give me ~30h.

Reply:
Hang on, wait, what is this even doing? `npmConfigHook` already runs after the defined `postPatch`...

Reply:
Check L67, they are making sure that they can evenlop the hook with pushd/popd to fix the sourceRoot.

Reply:
> Hang on, wait, what is this even doing? `npmConfigHook` already runs after the defined `postPatch`...

The pattern removes npmConfigHook from postPatchHooks.

Reply:
I think we could probably add an `npmRoot` kinda like `cargoRoot`. Would you like for me to work on that @winterqt?

Reply:
> I think we could probably add an `npmRoot` kinda like `cargoRoot`. Would you like for me to work on that @winterqt?

That's exactly what @mweinelt is asking for, yeah. I have a branch somewhere that does this, but it's simple enough to implement that if you can do it in the next few hours, go for it. (Else, I will some time tomorrow -- I really don't mind.)

Reply:
> (alternatively provide pre and post hooks for npmConfigHook)

This is already a thing, FYI -- `prePatch` and `postPatch` both run *before* `npmConfigHook`.

Reply:
Heh. The idea was to have something that runs before and after `npmConfigHook`, just like what's currently done in this PR. :laughing: